### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by jingwei99**

### CODE REVIEW

File: redwood-lazylayout-compose/src/commonMain/kotlin/app/cash/redwood/lazylayout/compose/LazyListState.kt [hunk 1]
```diff
@@ -43,7 +44,26 @@ public class LazyListState(
...
+    /**
+     * The default [Saver] implementation for [LazyListState].
+     */
+    public val Saver: Saver<LazyListState, *> = Saver(
+      save = {
+        println("**** saved index: ${it.firstVisibleItemIndex}")
+        it.firstVisibleItemIndex
+      },
+      restore = {
+        println("**** restored index: $it")
```
Comment (jingwei99) [inline redwood-lazylayout-compose/src/commonMain/kotlin/app/cash/redwood/lazylayout/compose/LazyListState.kt:62]: log

File: redwood-lazylayout-compose/src/commonMain/kotlin/app/cash/redwood/lazylayout/compose/LazyListState.kt [hunk 2]
```diff
@@ -43,7 +44,26 @@ public class LazyListState(
...
     scrollToItemTriggeredId++
   }
+
+  public companion object {
+    /**
+     * The default [Saver] implementation for [LazyListState].
+     */
+    public val Saver: Saver<LazyListState, *> = Saver(
+      save = {
+        println("**** saved index: ${it.firstVisibleItemIndex}")
```
Comment (jingwei99) [inline redwood-lazylayout-compose/src/commonMain/kotlin/app/cash/redwood/lazylayout/compose/LazyListState.kt:58]: log

File: redwood-lazylayout-compose/src/commonMain/kotlin/app/cash/redwood/lazylayout/compose/LazyListState.kt [hunk 3]
```diff
@@ -43,7 +44,26 @@ public class LazyListState(
   public fun scrollToItem(
     index: Int,
   ) {
+    println("**** scroll to item triggered with $index")
```
Comment (jingwei99) [inline redwood-lazylayout-compose/src/commonMain/kotlin/app/cash/redwood/lazylayout/compose/LazyListState.kt:47]: log

File: samples/emoji-search/presenter/src/commonMain/kotlin/com/example/redwood/emojisearch/presenter/EmojiSearch.kt [hunk 4]
```diff
@@ -123,9 +123,18 @@ private fun LazyColumn(
...
+//    lazyListState.scrollToItem(lazyListState.firstVisibleItemIndex)
+//  }
 
-  LaunchedEffect(searchTerm) {
-    lazyListState.scrollToItem(0)
+  LaunchedEffect(allEmojis.size) {
+    if (allEmojis.size > lazyListState.firstVisibleItemIndex) {
+      println("**** all emojis size: ${allEmojis.size}")
+      println("**** first visible item index: ${lazyListState.firstVisibleItemIndex}")
+      lazyListState.scrollToItem(lazyListState.firstVisibleItemIndex)
```
Comment (jingwei99) [inline samples/emoji-search/presenter/src/commonMain/kotlin/com/example/redwood/emojisearch/presenter/EmojiSearch.kt:136]: log

output of logs for the following events
1) scroll down on the list
2) click on an emoji, UI will navigate away to chrome 
3) click back, UI will navigate back to app 

Screenshot 2023-10-12 at 10 38 36 AM

but list stays on top (doesn't scroll to position)

**Comment by swankjesse** [comment]
Obsoleted by https://github.com/cashapp/redwood/pull/1619

**Pull request closed without merging**